t/resty/sanity.t

```diff
@@ -71,3 +71,26 @@ hello
 world
 --- err
 --- ret: 0
+
+
+
+=== TEST 6: catch interrupted fread during file readall (GH issue #35)
+--- src
+local ffi = require "ffi"
+
+ffi.cdef [[
+    int getpid(void);
+]]
+
+local pid = ffi.C.getpid()
+
+local signal = 17
+local platform = assert(io.popen("uname"):read("*l"))
+if platform == "Darwin" then
+    signal = 0
+end
+
+local cmd = string.format("kill -%d %d && sleep 0.1", signal, pid)
+assert(io.popen(cmd):read("*a"))
```

Comment:
It would be great if we could add a test case for the case when the underlying `read()` syscall encounters an error other than EINTR. But it seems tricky without the help of special instrumenting facility like gdb or systemtap :(

Reply:
I think I found a way! See the latest patch.